cognite/client/cognite_client.py

```diff
@@ -112,8 +115,8 @@ def __init__(
 
         self._requests_session = self._requests_retry_session()
 
-        self._project = project
-        if project is None:
+        self._project = project or thread_local_project
```

Comment:
Maybe this should follow the decision of which api-key to use? Might be annoying (or nice feature?) if you have project in thread local or environment, but then specify api-key as argument (but not project). Then it will not auto fetch project, but instead use thread local or environment - which is not what you expect and a 403 will be puzzling.

Reply:
If we were to use environment project, then I sort of agree with you. But since thread-local config will only be used by users who really know what they are doing, normal users will only override the autofetching by specifying the project as a parameter. So I think it is difficult to get in a situation where you get an unexpected 401.